dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java

```diff
@@ -15,20 +15,36 @@
  */
 package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
 
-import org.springframework.batch.item.ItemProcessor;
-
-import uk.ac.ebi.eva.commons.core.models.IVariant;
-import uk.ac.ebi.eva.commons.core.models.VariantCoreFields;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
 import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
 
-public class SubSnpCoreFieldsToVariantProcessor implements ItemProcessor<SubSnpCoreFields, IVariant> {
+/**
+ * TODO Create copy of this class for variants submitted by EVA, which don't need a nested VariantSourceEntry object
+ */
+public class SubSnpCoreFieldsToVariantProcessor extends SubSnpCoreFieldsToCoreVariantProcessor {
 
-    @Override
-    public IVariant process(SubSnpCoreFields subSnpCoreFields) throws Exception {
-        VariantCoreFields variantCoreFields = subSnpCoreFields.getVariantCoreFields();
-        
-        return null;
+    public static final String DBSNP_BUILD_KEY = "dbSNP build";
+
+    private final String dbsnpBuild;
+
+    private final String batch;
+
+    public SubSnpCoreFieldsToVariantProcessor(int dbsnpBuild, int batch) {
```

Comment:
Now I think it would be more illustrative to use the batch name than the number. In fact, [the samples reader retrieves several batch attributes](https://github.com/EBIvariation/eva-tools/blob/develop/dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SampleReader.java#L58-L60) but the variants reader doesn't.